src/overlays/actors/ovl_En_Talk_Gibud/z_en_talk_gibud.c

```diff
@@ -28,166 +83,1063 @@ const ActorInit En_Talk_Gibud_InitVars = {
+    func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_WALK);
+    this->actor.speedXZ = 0.4f;
+    if (this->actionFunc == EnTalkGibud_AttemptPlayerStun) {
+        this->playerStunWaitTimer = 80;
+    } else {
+        this->playerStunWaitTimer = 20;
+    }
+    this->actionFunc = EnTalkGibud_WalkToPlayer;
+}
+
+void EnTalkGibud_WalkToPlayer(EnTalkGibud* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 pad;
+
+    Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xFA);
+    this->actor.world.rot = this->actor.shape.rot;
+    Math_SmoothStepToS(&this->headRotation.y, 0, 1, 100, 0);
+    Math_SmoothStepToS(&this->upperBodyRotation.y, 0, 1, 100, 0);
+    if (EnTalkGibud_PlayerInRangeWithCorrectState(this, globalCtx) && Actor_IsActorFacingLink(&this->actor, 0x38E3)) {
+        if (this->grabWaitTimer == 0 && this->actor.xzDistToPlayer <= 45.0f) {
+            player->actor.freezeTimer = 0;
+            if (gSaveContext.playerForm == PLAYER_FORM_GORON || gSaveContext.playerForm == PLAYER_FORM_DEKU) {
+                // If the Gibdo/Redead tries to grab Goron or Deku Link, it will fail to
+                // do so. It will appear to take damage and shake its head side-to-side.
+                EnTalkGibud_SetupGrabFail(this);
+            } else if (globalCtx->grabPlayer(globalCtx, player)) {
+                EnTalkGibud_SetupGrab(this);
+            }
+        } else {
+            if (this->playerStunWaitTimer == 0) {
+                player->actor.freezeTimer = 40;
+                this->playerStunWaitTimer = 60;
+                func_8013ECE0(this->actor.xzDistToPlayer, 255, 20, 150);
+                func_80123E90(globalCtx, &this->actor);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_AIM);
+            } else {
+                this->playerStunWaitTimer--;
+            }
+        }
+    } else if (this->grabWaitTimer == 0 && this->actor.xzDistToPlayer <= 45.0f) {
+        EnTalkGibud_SetupWalkToHome(this);
+    } else if (EnTalkGibud_PlayerOutOfRange(this, globalCtx)) {
+        EnTalkGibud_SetupWalkToHome(this);
+    }
+    if (this->grabWaitTimer > 0) {
+        this->grabWaitTimer--;
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 10.0f) || Animation_OnFrame(&this->skelAnime, 22.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_WALK);
+    } else if (!(globalCtx->gameplayFrames & 0x5F)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_CRY);
+    }
+}
+
+void EnTalkGibud_SetupGrab(EnTalkGibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_GRAB_START);
+    this->grabDamageTimer = 0;
+    this->actor.flags &= ~1;
+    this->grabState = EN_TALK_GIBUD_GRAB_START;
+    this->actionFunc = EnTalkGibud_Grab;
+}
+
+void EnTalkGibud_Grab(EnTalkGibud* this, GlobalContext* globalCtx) {
+    Player* player2 = GET_PLAYER(globalCtx);
+    Player* player = player2;
```

Comment:
So all the `stateFlag` stuff in the second part of this function *only* matches if there are two different temps of Player, because it *has* to use the "second" temp. However, you can do this and it still matches:
```c
    Actor* playerActor = &GET_PLAYER(globalCtx)->actor;
    Player* player = (Player*)playerActor;
```

I'm guessing what happened here is they didn't use a macro and just manually got an `Actor*` instance of Player from `globalCtx`, then casted it to `Player*` in a second temp. The way I did it looks a bit silly, so I'll leave this unresolved in case people don't like it.